### PR TITLE
Issue-1852: Upgrade our JUnit dependencies from 5.4.0 to 5.7.0

### PR DESCRIPTION
**Pull request by ariffebr**
- [x] upgrade junit version to 5.7.0
- [x] exlcude junit under org.codehaus.mojo.appassembler:appassembler-model

**cc:** @carlspring  @sbespalov  @steve-todorov

### CODE REVIEW

**Comment by carlspring** [comment]
Hi!

Thank you for your contribution!

Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

Also, please, feel free to join our [chat channel](https://chat.carlspring.org/), if you'd like to learn more about the project and/or like to find out what else you could help with.

Kind regards,

Martin

**Comment by carlspring** [comment]
@ptirador ,

Could you please review? Thanks!

**Comment by ariffebr** [comment]
this PR also for [Issue-1870: Upgrade the maven-surefire-plugin and maven-failsafe-plugin to 3.0.0-M5](https://github.com/strongbox/strongbox/issues/1870)

File: pom.xml [hunk 1]
```diff
@@ -158,7 +158,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
-                    <version>2.22.1</version>
+                    <version>3.0.0-M5</version>
```
Comment (ptirador) [inline pom.xml:161]: Why this change? This is not the scope of this issue. Please roll this change back, and apply it in a new PR with the corresponding issue.

File: pom.xml [hunk 2]
```diff
@@ -200,7 +200,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-failsafe-plugin</artifactId>
-                    <version>2.22.1</version>
+                    <version>3.0.0-M5</version>
```
Comment (ptirador) [inline pom.xml:203]: Why this change? This is not the scope of this issue. Please roll this change back, and apply it in a new PR with the corresponding issue.

**Review by ptirador — CHANGES_REQUESTED** [review]

**Comment by carlspring** [comment]
Hi!

Thank you for your contribution!

Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

Also, please, feel free to join our [chat channel](https://chat.carlspring.org/), if you'd like to learn more about the project and/or like to find out what else you could help with.

Kind regards,

Martin

File: pom.xml (hunk 1 above)
Comment (carlspring) [reply pom.xml:161]: Yeah, I think it would be better to have these plugin updates handled in a separate pull request.

**Comment by carlspring** [comment]
Also, is there an associated (test) pull request for the `strongbox` project?

**Comment by ptirador** [comment]
> Also, is there an associated (test) pull request for the `strongbox` project?

Yes, I agree with @carlspring. You'll need to follow these steps:
1. Change the version of `strongbox-parent` to be `1.0-issue-1852-dep-upgrade-SNAPSHOT` and wait for the Jenkins build to complete - it will also deploy this version into a temporary repository.
2. Update the parent version of `strongbox` so that it is `1.0-issue-1852-dep-upgrade-SNAPSHOT`, which will then use this version in the builds. 
2.1. You can do that with `mvn versions:update-parent -DallowSnapshots=true -DparentVersion=1.0-issue-1852-dep-upgrade-SNAPSHOT`.

**Comment by carlspring** [comment]
@ariffebr,

Have you sent us your signed ICLA?

**Comment by carlspring** [comment]
@ariffebr ,

As you haven't signed the ICLA, (and responded), or created an associated test pull request for this in the strongbox project, I'm afraid I'll have to close this and hand the task over the @ptirador .

Feel free to pick another task, if you like.

Thank you or you efforts!
